braidpool-primitives/src/braid/mod.rs

```diff
@@ -85,6 +110,120 @@ impl Braid {
 }
 
 impl Braid {
+    // All pub(crate) functions go here!
+    #[inline]
+    pub(crate) fn load_bead_from_hash(&self, bead_hash: BeadHash) -> Result<&Bead, BeadLoadError> {
```

Comment:
This should be done by Bead serialization, and the db work should go into the Node.